.github/workflows/run-tests.yml

```diff
@@ -264,6 +265,12 @@ jobs:
           TEST_ARGS: ${{ needs.set-up-single-test.outputs.single_test_args }}
           TEST_TIMEOUT: ${{ needs.set-up-single-test.outputs.test_timeout }}
 
+      - name: Generate crash report
+        if: failure()
+        run: | # if the tests failed, we would expect one JUnit XML report per attempt; otherwise it must have crashed
+          [ "$(find .testoutput -maxdepth 1 -name 'junit.*.xml' | wc -l)" -lt "$TEST_ATTEMPTS" ] && 
+            CRASH_REPORT_NAME="$GITHUB_JOB" make report-test-crash
```

Comment:
New step that checks if there was a crash; and then emits the JUnit crash report file.